### PR TITLE
Mldb 2092 plugin autoload js and py

### PR DESCRIPTION
**Pull request by FinchPowers**

### CODE REVIEW

File: container_files/public_html/doc/builtin/plugins/Plugins.md [hunk 1]
```diff
@@ -16,8 +16,10 @@ Plugins are loaded into MLDB via a [REST API call](PluginConfig.md).
 
 MLDB will load up system plugins from `/opt/mldb/plugins` automatically on startup.
 
-MLDB will scan the directory in `/mldb_data/plugins/autoload` and load any plugins it
-finds under that directory.
+MLDB will scan the directory in `/mldb_data/plugins/autoload`, and for each file named `mldb_plugin.json` found,
```
Comment (FinchPowers) [inline container_files/public_html/doc/builtin/plugins/Plugins.md]: directory or directories?

File: container_files/public_html/doc/builtin/plugins/Plugins.md [hunk 2]
```diff
@@ -16,8 +16,10 @@ Plugins are loaded into MLDB via a [REST API call](PluginConfig.md).
...
+will autoload the plugin described by it. `mldb_plugin.json` must contain the same JSON payload that would be used to POST or
+PUT the plugin to MLDB.
```
Comment (guyd) [inline container_files/public_html/doc/builtin/plugins/Plugins.md]: Consider adding a sample payload.

File: server/mldb_server.cc [hunk 3]
```diff
@@ -410,24 +411,37 @@ scanPlugins(const std::string & dir_)
...
+                         manifest.config.type == "javascript") {
+                    auto config = manifest.config.params.convert<PluginResource>();
+                    config.address = dir;
+                    manifest.config.params = config;
+                    auto plugin = plugins->obtainEntitySync(
+                        manifest.config, nullptr /* on progress */);
+                }
+                else {
+                    throw HttpReturnException(
+                        500, "unknown plugin type to autoload at " + dir);
```
Comment (guyd) [inline server/mldb_server.cc]: Can really throw an HttpReturnException here?  There are no one to catch the response!  I suggest we log an error message instead and continue the processing.

File: container_files/public_html/doc/builtin/plugins/Plugins.md (hunk 1 above)
Comment (guyd) [inline container_files/public_html/doc/builtin/plugins/Plugins.md]: May be _MLDB will recursively scan the directory `/mldb_data/plugins/autoload`.

**Comment by guyd** [comment]
+1

File: container_files/public_html/doc/builtin/plugins/Plugins.md [hunk 4]
```diff
@@ -16,8 +16,21 @@ Plugins are loaded into MLDB via a [REST API call](PluginConfig.md).
...
+```
+{
+    "config" : {
+        "id" : "demo_plugin",
+        "type" : "python",
+        "params" : { }
+    }
+}
+```
+`params` can be left empty as MLDB will automatically set the "adress" value to the path where the plugin was found.
```
Comment (FinchPowers) [inline container_files/public_html/doc/builtin/plugins/Plugins.md]: @guyd what do you think about it now?

File: container_files/public_html/doc/builtin/plugins/Plugins.md [hunk 5]
```diff
@@ -16,10 +16,21 @@ Plugins are loaded into MLDB via a [REST API call](PluginConfig.md).
 
 MLDB will load up system plugins from `/opt/mldb/plugins` automatically on startup.
 
-MLDB will scan the directory in `/mldb_data/plugins/autoload`, and for each file named `mldb_plugin.json` found,
-will autoload the plugin described by it. `mldb_plugin.json` must contain the same JSON payload that would be used to POST or
-PUT the plugin to MLDB.
-
+MLDB will recursively scan the directory `/mldb_data/plugins/autoload`, and for each file named `mldb_plugin.json` found,
+will autoload the plugin described by it. `mldb_plugin.json` must contain the a JSON payload with a single top key `config`
```
Comment (guyd) [inline container_files/public_html/doc/builtin/plugins/Plugins.md]: ~the~

File: container_files/public_html/doc/builtin/plugins/Plugins.md (hunk 4 above)
Comment (guyd) [inline container_files/public_html/doc/builtin/plugins/Plugins.md]: +1 beside an extra _the_ to remove

**Comment by FinchPowers** [comment]
verbal +1

**Pull request merged**